src/_zkapauthorizer/tahoe.py

```diff
@@ -430,18 +450,16 @@ async def download(self, outpath, cap, child_path):
         if child_path is not None:
             for p in child_path:
                 if cap.startswith("URI:DIR2"):
-                    cap = d[p]
+                    cap = d.children[p]
                     d = self._grid.download(cap)
                 else:
                     raise TahoeAPIError(
                         "get", DecodedURL(), 400, _no_children_message.format(path=p)
                     )
-        if isinstance(d, dict):
-            # It is a directory.  Encode it somehow so it fits in the file.
-            # This is not the same encoding as Tahoe-LAFS itself uses for
-            # directories.
-            d = dumps_utf8(d)
-        outpath.setContent(d)
+        if isinstance(d, bytes):
+            outpath.setContent(d)
+        else:
+            raise ValueError(f"Cannot download non-data capability ({cap[:7]})")
```

Comment:
It was basically wrong to let this API download a dircap before - you can't do that with the real tahoe client (or, you can, but you get data in a totally different format than we were giving out).